tests/test_config_entries.py

```diff
@@ -3330,3 +3330,110 @@ async def test_reauth(hass):
     entry2.async_start_reauth(hass, {"extra_context": "some_extra_context"})
     await hass.async_block_till_done()
     assert len(hass.config_entries.flow.async_progress()) == 2
+
+
+async def test_wait_for_loading_entry(hass):
+    """Test waiting for entry to be set up."""
+
+    entry = MockConfigEntry(title="test_title", domain="test")
+
+    mock_setup_entry = AsyncMock(return_value=True)
+    mock_integration(hass, MockModule("test", async_setup_entry=mock_setup_entry))
+    mock_entity_platform(hass, "config_flow.test", None)
+
+    await entry.async_setup(hass)
+    await hass.async_block_till_done()
+
+    flow = hass.config_entries.flow
+
+    async def _load_entry():
+        # Mock config entry
+        assert await async_setup_component(hass, "test", {})
+
+    entry = MockConfigEntry(title="test_title", domain="test")
```

Comment:
Why do we overwrite the config entry with another entry with the same details?

Reply:
See https://github.com/home-assistant/core/pull/81771 (second commit)